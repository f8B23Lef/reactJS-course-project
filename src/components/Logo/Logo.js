import React from 'react';
import PropTypes from 'prop-types';
import logo from '../../assets/images/netflixLogo.png';
import './Logo.scss';

export default function Logo({ src }) {
  return (
    <img className='logo' src={src} alt='netflix logo' />
  );
}

Logo.propTypes = {
  src: PropTypes.string,
};

Logo.defaultProps = {
  src: logo,
};
