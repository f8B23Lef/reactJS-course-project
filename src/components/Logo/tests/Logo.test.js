import React from 'react';
import { render, screen } from '@testing-library/react';
import Logo from '../Logo';

describe('Logo', () => {
  it('renders Logo snapshot', () => {
    const { asFragment } = render(<Logo src='/logo' />);
    expect(asFragment(<Logo src='/logo' />)).toMatchSnapshot();
  });

  it('renders image with specific class and path', () => {
    render(<Logo src='/logo' />);
    const imageElement = screen.getByAltText('netflix logo');

    expect(imageElement).toHaveAttribute('src', '/logo');
    expect(imageElement).toHaveClass('logo');
  });
});
