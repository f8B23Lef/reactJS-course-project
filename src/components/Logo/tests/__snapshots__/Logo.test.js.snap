// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders Logo snapshot 1`] = `
<DocumentFragment>
  <img
    alt="netflix logo"
    class="logo"
    src="/logo"
  />
</DocumentFragment>
`;
